druid/Dockerfile

```diff
@@ -4,13 +4,76 @@
 # Ignoring DL4006 globally because we inherit the SHELL from our base image
 # hadolint global ignore=DL3038,DL4006
 
-FROM stackable/image/java-base
+# hadolint ignore=DL3006
+FROM stackable/image/java-devel as druid-builder
 
 ARG PRODUCT
 ARG JACKSON_DATAFORMAT_XML
 ARG STAX2_API
 ARG WOODSTOX_CORE
 ARG AUTHORIZER
+
+RUN microdnf update && \
+    microdnf install \
+    # Required to install pyyaml
+    python-pip && \
+    microdnf clean all && \
+    rm -rf /var/cache/yum && \
+    # pyyaml is required for the compile Druid
+    pip install --no-cache-dir pyyaml==6.0.1
+
+USER stackable
+WORKDIR /stackable
+
+RUN curl --fail -L "https://repo.stackable.tech/repository/packages/druid/apache-druid-${PRODUCT}-src.tar.gz" | tar -xzC . && \
+    cd apache-druid-${PRODUCT}-src && \
+    mvn clean install -Pdist -DskipTests -Dmaven.javadoc.skip=true && \
+    tar -xzf /stackable/apache-druid-${PRODUCT}-src/distribution/target/apache-druid-${PRODUCT}-bin.tar.gz && \
+    mv /stackable/apache-druid-${PRODUCT}-src/apache-druid-${PRODUCT} /stackable/apache-druid-${PRODUCT} && \
+    rm -rf /stackable/apache-druid-${PRODUCT}-src \
+    /stackable/apache-druid-${PRODUCT}/quickstart
+
+    # Install the Prometheus emitter extension. This bundle contains the emitter and all jar dependencies.
+RUN curl --fail "https://repo.stackable.tech/repository/packages/druid/druid-prometheus-emitter-${PRODUCT}.tar.gz" | tar -xzC /stackable/apache-druid-${PRODUCT}/extensions && \
+    # Install OPA authorizer extension.
+    curl --fail "https://repo.stackable.tech/repository/packages/druid/druid-opa-authorizer-${AUTHORIZER}.tar.gz" | tar -xzC /stackable/apache-druid-${PRODUCT}/extensions && \
+    # Install jackson-dataformat-xml, stax2-api, and woodstox-core which are required for logging, and remove stax-ex.
+    rm /stackable/apache-druid-${PRODUCT}/lib/stax-ex-*.jar && \
+    curl --fail -L -o /stackable/apache-druid-${PRODUCT}/lib/jackson-dataformat-xml-${JACKSON_DATAFORMAT_XML}.jar \
+        "https://repo.stackable.tech/repository/packages/jackson-dataformat-xml/jackson-dataformat-xml-${JACKSON_DATAFORMAT_XML}.jar" && \
+    curl --fail -L -o /stackable/apache-druid-${PRODUCT}/lib/stax2-api-${STAX2_API}.jar \
+        "https://repo.stackable.tech/repository/packages/stax2-api/stax2-api-${STAX2_API}.jar" && \
+    curl --fail -L -o /stackable/apache-druid-${PRODUCT}/lib/woodstox-core-${WOODSTOX_CORE}.jar \
+        "https://repo.stackable.tech/repository/packages/woodstox-core/woodstox-core-${WOODSTOX_CORE}.jar"
+
+# ===
+# Mitigation for CVE-2021-44228 (Log4Shell)
+# This variable is supported as of Log4j version 2.10 and
+# disables the vulnerable feature
+ENV LOG4J_FORMAT_MSG_NO_LOOKUPS=true
```

Comment:
This should go into the main image.

Reply:
Done https://github.com/stackabletech/docker-images/pull/684/commits/f33bc215feafade57b11ff1727a1c6ae3ef2d1f2